implementations/rust/ockam/ockam_node/src/storage/database/sqlx_types.rs

```diff
@@ -0,0 +1,200 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+
+use ockam_core::Address;
+use sqlx::database::HasArguments;
+use sqlx::encode::IsNull;
+use sqlx::{Database, Encode, Sqlite, Type};
+use time::OffsetDateTime;
+
+/// This enum represents the set of types that we currently support in our database
+/// Since we support only Sqlite at the moment, those types are close to what is supported by Sqlite:
+/// https://www.sqlite.org/datatype3.html
+///
+/// The purpose of this type is to ease the serialization of data types in Ockam into data types in
+/// our database. For example, if we describe how to translate an `Identifier` into some `Text` then
+/// we can use the `Text` as a parameter in a sqlx query.
+///
+/// Note: see the `ToSqlxType` trait and its instances for how the conversion is done
+///
+pub enum SqlxType {
+    /// This type represents text in the database
+    Text(String),
+    /// This type represents arbitrary bytes in the database
+    Blob(Vec<u8>),
+    /// This type represents ints, signed or unsigned
+    Integer(i64),
+    /// This type represents floats
+    #[allow(unused)]
+    Real(f64),
+}
+
+/// The SqlType implements the Type<Sqlite> trait from sqlx to allow its values to be serialized
```

Comment:
Could you please elaborate on how this is used? I have problems understanding why we return `Vec<u8>::type_info()` here. (I understand that it's a static method and we can't return a different type info depending on the self value)

Reply:
My understanding is that this is just a default value in this case, which is overridden by the `Encode` instance on line 42. This is what the doc says about this function:
```
When binding arguments, this is used to tell the database what is about to be sent; which, the database then uses to guide query plans. This can be overridden by Encode::produces
```